tasks/coverage.py

```diff
@@ -119,11 +119,16 @@ def upload_to_codecov(
     distro_tag = get_distro()
     codecov_binary = "codecov" if platform.system() != "Windows" else "codecov.exe"
 
-    with gitlab_section("Upload coverage reports to Codecov", collapsed=True):
+    coverage_cache_section = (
+        "Applying missing coverage cache from S3." if pull_coverage_cache else "Uploading coverage files to S3."
+    )
+    with gitlab_section(coverage_cache_section, collapsed=True):
         if pull_coverage_cache:
             apply_missing_coverage(ctx, from_commit_sha=get_main_parent_commit(ctx), debug=debug)
         if push_coverage_cache:
             upload_coverage_to_s3(ctx)
```

Comment:
nit: I'd have nested the section in the CI and avoid the `coverage_cache_section` because it took me a bit of time to understand why you did that and if one day we add another `if` inside the section we'll need to modify it

Reply:
Good point ! Fixed with [60eb80f](https://github.com/DataDog/datadog-agent/pull/26991/commits/60eb80feb57dafe74115fa78ef4060e8ba6c2f06).